### PR TITLE
MISC: Added NS function closeTail to close tail windows

### PR DESCRIPTION
**Pull request by Undeemiss**
Old, and now inaccurate description of this PR, kept for posterity (See later comment for new description):

Attempted to implement `ns.closeTail()`, `ns.closeTail(pid)`, and `ns.closeTail(fn, hostname, ...args)`. Currently, `ns.closeTail()` works, `ns.closeTail(pid)` works if the given pid points to a script that's still running, and `ns.closeTail(fn, hostname, ...args)` works. It has not been tested extensively, and has not been formatted/linted because it's a WIP.

Related to #3585

### CODE REVIEW

File: src/ui/React/LogBoxManager.tsx [hunk 1]
```diff
@@ -131,6 +142,8 @@ function LogWindow(props: IProps): React.ReactElement {
     return () => clearInterval(id);
   }, []);
 
+  //TODO Not actually a todo I just want this file to show up as changed
+  //so I can find it easier later
```
Comment (hydroflame) [inline src/ui/React/LogBoxManager.tsx:146]: rm

**Comment by Undeemiss** [comment]
New description of PR:

Implemented NS function `closeTail(pid?: number)`, which closes any open tail window with the given pid. If no pid is given, closes any open tail window of the calling script. 

Tested by ensuring it properly handles the no input case, as well as pids corresponding to dead scripts, running scripts, pids that have never been used, and pids that correspond to tail windows that have run a new copy of the same script.
Notably, it does nothing, not even giving an error, when given a pid that does not correspond to a script associated with a tail window, and tail windows are only associated with the most recent pid (i.e. if you run some script test.js with pid 1, close it via a tail window, and rerun it with pid 2, the tail window will only correspond to pid 2)

Related to #3585, but does not satisfy it completely

File: src/NetscriptFunctions.ts [hunk 2]
```diff
@@ -988,6 +988,15 @@ export function NetscriptFunctions(workerScript: WorkerScript): NS {
 
       LogBoxEvents.emit(runningScriptObj);
     },
+    closeTail: function (pid?: number): void {
```
Comment (hydroflame) [inline src/NetscriptFunctions.ts:991]: `pid?: unknown = workerScript.pid`

then call `helper.number`

Comment (Undeemiss) [reply src/NetscriptFunctions.ts]: Done, I think. It's not exactly what you suggested, but it matches the way another function right next to it works, and doing what you gave me exactly just gives a syntax error.

Comment (hydroflame) [reply src/NetscriptFunctions.ts]: The question mark in my example should not have been there

**Pull request merged**